zipline/utils/events.py

```diff
@@ -336,28 +336,39 @@ def __init__(self, offset=None, **kwargs):
             datetime.timedelta(minutes=1),  # Defaults to the first minute.
         )
 
-        self._next_period_start = None
-        self._next_period_end = None
+        self._period_start = None
+        self._period_end = None
+        self._period_close = None
 
         self._one_minute = datetime.timedelta(minutes=1)
 
     def calculate_dates(self, dt, env):
         # given a dt, find that day's open and period end (open + offset)
-        self._next_period_start = env.get_open_and_close(dt)[0]
-        self._next_period_end = \
-            self._next_period_start + self.offset - self._one_minute
+        self._period_start = env.get_open_and_close(dt)[0]
```

Comment:
this is doing an extra `get_open_and_close` unnecessarily.

you can probably do:
```
self._period_start, self._period_close = env.get_open_and_close(dt)
```